src/Config/FileConfig.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * @copyright 2017 Hostnet B.V.
+ */
+declare(strict_types=1);
+
+namespace Hostnet\Component\Resolver\Config;
+
+use Hostnet\Component\Resolver\File;
+use Hostnet\Component\Resolver\Import\Nodejs\Executable;
+use Hostnet\Component\Resolver\Plugin\PluginInterface;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Symfony\Component\EventDispatcher\EventDispatcher;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+final class FileConfig implements ConfigInterface
+{
+    private $config_file;
+
+    private $config_file_contents;
+
+    private $dispatcher;
+
+    private $logger;
+
+    private $plugins;
+
+    private $dev;
+
+    /**
+     * @param string $config_file
+     * @param PluginInterface[] $plugins
+     * @param bool $dev
+     * @param EventDispatcherInterface $dispatcher
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        string $config_file,
+        array $plugins = [],
+        bool $dev = false,
+        EventDispatcherInterface $dispatcher = null,
+        LoggerInterface $logger = null
+    ) {
+        $this->config_file          = $config_file;
+        $this->config_file_contents = json_decode(file_get_contents($this->config_file), true);
```

Comment:
Not sure how safe this is, what if there's invalid json or the file does not exist?